embedding/cpp/include/flutter_engine.h

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_TIZEN_EMBEDDING_CPP_INCLUDE_FLUTTER_ENGINE_H_
+#define FLUTTER_TIZEN_EMBEDDING_CPP_INCLUDE_FLUTTER_ENGINE_H_
+
+#include <app_control.h>
+#include <flutter/plugin_registry.h>
+#include <flutter_tizen.h>
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+// The engine for flutter execution.
+class FlutterEngine : public flutter::PluginRegistry {
+ public:
+  virtual ~FlutterEngine();
+  static std::unique_ptr<FlutterEngine> Create(
+      const std::optional<std::string>& dart_entrypoint = std::nullopt,
+      const std::optional<std::vector<std::string>>& dart_entrypoint_args =
+          std::nullopt);
+  static std::unique_ptr<FlutterEngine> Create(
+      const std::string& assets_path, const std::string& icu_data_path,
+      const std::string& aot_library_path,
+      const std::optional<std::string>& dart_entrypoint = std::nullopt,
+      const std::optional<std::vector<std::string>>& dart_entrypoint_args =
+          std::nullopt);
+
+  // Prevent copying.
+  FlutterEngine(FlutterEngine const&) = delete;
+  FlutterEngine& operator=(FlutterEngine const&) = delete;
+
+  // Starts running the engine.
+  bool Run();
+
+  // Terminates the running engine.
+  void Shutdown();
+
+  // Whether the engine is running.
+  bool IsRunning() { return is_running_; }
+
+  // Resumes the engine.
+  //
+  // This method notifies the running Flutter app that it is "resumed" as per
+  // the Flutter app lifecycle.
+  void NotifyAppIsResumed();
+
+  // Pauses the engine.
+  //
+  // This method notifies the running Flutter app that it is "inactive" as per
+  // the Flutter app lifecycle.
+  void NotifyAppIsInactive();
+
+  // Stops the engine.
+  //
+  // This method notifies the running Flutter app that it is "paused" as per
+  // the Flutter app lifecycle.
+  void NotifyAppIsPaused();
+
+  // Detaches the engine.
+  //
+  // This method notifies the running Flutter app that it is "detached" as per
+  // the Flutter app lifecycle.
+  void NotifyAppIsDetached();
+
+  // Notifies that the host app received an app control.
+  //
+  // This method sends the app control to Flutter over the "app control event
+  // channel".
+  void NotifyAppControl(app_control_h app_control);
+
+  // Notifies that low memory warning.
```

Comment:
```suggestion
  // Notifies that a low memory warning has been received.
```

Reply:
Thank you for your suggestion. I applied it.